src/Lecture1.hs

```diff
@@ -98,7 +107,10 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> [Char] -> [Char]
+subString start end str
+    | end < 0 = []
+    | otherwise = take (end-(max 0 start)+1) (drop (max 0 start) str)
```

Comment:
Usually, it's recommended to put spaces around binary operators 👾 
```suggestion
    | otherwise = take (end - (max 0 start) + 1) (drop (max 0 start) str)
```